packages/core-data/src/fetch/__experimental-fetch-link-suggestions.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+import { decodeEntities } from '@wordpress/html-entities';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Filters the search by type
+ *
+ * @typedef { 'post' | 'term' | 'post-format' } WPLinkSearchType
+ */
+
+/**
+ * @typedef WPLinkSearchOptions
+ *
+ * @property {boolean}             [isInitialSuggestions] Displays initial search suggestions, when true.
+ * @property {WPLinkSearchType}    [type]                 Filters by search type.
+ * @property {string}              [subtype]              Slug of the post-type or taxonomy.
+ * @property {number}              [page]                 Which page of results to return.
+ * @property {number}              [perPageArg]           Search results per page.
+ */
+
+/**
+ * @typedef WPLinkSearchResult
+ *
+ * @property {number} id     Post or term id.
+ * @property {string} url    Link url.
+ * @property {string} title  Title of the link.
+ * @property {string} type   The taxonomy or post type slug or type URL.
+ */
+
+/**
+ * Fetches link suggestions from the API.
+ *
+ * @async
+ * @typedef {( searchText: string, searchOptions: WPLinkSearchOptions ) => Promise< WPLinkSearchResult[] >} WPFetchLinkSuggestions
+ */
+
+/**
+ * Returns a function that when invoked, fetches link suggestions from the API.
+ *
+ * @param {Object}   [editorSettings]
+ * @param {boolean}  [editorSettings.disablePostFormats]
+ *
+ * @example
+ * ```js
+ * import { __experimentalFetchLinkSuggestions as createFetchLinkSuggestions } from '@wordpress/core-data';
+ *
+ * //...
+ *
+ * export function initialize( id, settings ) {
+ *
+ * settings.__experimentalFetchLinkSuggestions = createFetchLinkSuggestions(
+		settings
+	);
+ * ```
+ * @return { WPFetchLinkSuggestions } Function that fetches link suggestions
+ *
+ */
+const createFetchLinkSuggestions = ( {
+	disablePostFormats = false,
```

Comment:
Minor, but I think it's easier to understand options like this when they're framed positively instead of negatively because then you are less likely to end up in situations where there's a double negative. So `enablePostFormats` instead of `disablePostFormats`.